core/interop/src/loaders/functions.rs

```diff
@@ -0,0 +1,63 @@
+//! This module contains types that help create custom module loaders from functions.
+use boa_engine::module::{ModuleLoader, Referrer};
+use boa_engine::{Context, JsError, JsNativeError, JsResult, JsString, Module};
+
+/// Create a [`ModuleLoader`] from a function that takes a referrer and a path,
+/// and returns a [Module] if it exists. This is the simplest way to create a
+/// custom [`ModuleLoader`].
+///
+/// This function cannot be `async` and must be blocking. An `async` version of
+/// this code will likely exist as a separate function in the future.
+///
+/// `F` cannot be a mutable closure as it could recursively call itself.
+#[derive(Copy, Clone)]
+pub struct FnModuleLoader<F>(F, &'static str)
+where
+    F: Fn(&JsString) -> Option<Module>;
```

Comment:
Was thinking that maybe this should return `JsResult<Module>`. We could also have a separate loaded (`FallibleFnModuleLoader` or something), but it seems better to just have this be the fallible version already.

Reply:
I'm thinking here that this is the simplest way to declare a module loader, without even worry about the error type and message; either the module exists or it doesn't. Even if it doesn't parse, or any other kind of error, just say it cannot be found.

I could see a better `Option<Source>` though, and show errors on parsing after the function returns the source of the module. WDYT?